features/user/components/userProfile/ChangePasswordPanel.tsx

```diff
@@ -0,0 +1,189 @@
+'use client'
+
+import { LoadingSpinner } from '@/components/ui/LoadingSpinner'
+import { Button } from '@/components/ui/button'
+import { Card, CardContent, CardHeader } from '@/components/ui/card'
+import {
+	Form,
+	FormControl,
+	FormField,
+	FormItem,
+	FormLabel,
+	FormMessage,
+} from '@/components/ui/form'
+import { Input } from '@/components/ui/input'
+import React, { useEffect, useState, useTransition } from 'react'
+import { useForm } from 'react-hook-form'
+import { EditPasswordSchema } from '../../schemas'
+import { z } from 'zod'
+import bcrypt from 'bcryptjs'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { changePassword, getUserById } from '../../actions'
+import { signOut, useSession } from 'next-auth/react'
+import { useToast } from '@/components/ui/use-toast'
+
+type ChangePasswordData = z.infer<typeof EditPasswordSchema>
+
+export const ChangePasswordPanel = () => {
+	const [password, setPassword] = useState<string | null>('')
+	const [isFetchPasswordPending, startFetchPasswordTransition] = useTransition()
+
+	const form = useForm<ChangePasswordData>({
+		resolver: zodResolver(EditPasswordSchema),
+		defaultValues: {
+			password: '',
+			newPassword: '',
+			passwordConfirmation: '',
+		},
+	})
+	const { toast } = useToast()
+	const {
+		handleSubmit,
+		formState: { errors },
+		reset,
+	} = form
+
+	const { data: session } = useSession()
+	const currentUserId = session?.user?.id
+	useEffect(() => {
+		const fetchUser = async () => {
+			if (!currentUserId) {
+				return
+			}
+			const user = await getUserById(currentUserId)
+			if (user?.password === undefined) return
+			setPassword(user?.password)
+		}
+		startFetchPasswordTransition(() => {
+			fetchUser().then().catch()
+		})
+	}, [currentUserId])
+
+	const onSubmit = async (data: ChangePasswordData) => {
+		if (!password) return
+		const oldPasswordIsValid = await bcrypt.compare(data.password, password)
+		if (!oldPasswordIsValid) {
+			toast({
+				title: 'Ooops! Something went wrong',
+				description: 'Old password is incorrect',
+				variant: 'destructive',
+			})
+			reset()
+			return
+		}
+		if (data.newPassword === data.password) {
```

Comment:
Jak możesz patrzeć czy hasła są takie same? Jak używasz bcypta to nie możesz czegoś takiego zrobić, a jeżeli nie używasz to zmień to, nie można przechowywać haseł niezahashowanych. + To jest logika którą powinien obsługiwać backend